File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    dx, dy = [-1, 0, 1, 0], [0, 1, 0, -1]
    cur = board[h][w]
    answer = 0
    length = len(board)
    
    for i in range(4):
        cx, cy = h+dx[i], w+dy[i]
        if cx>=0 and cx<length and cy >=0 and cy<length:
            if cur == board[cx][cy]:
                answer+=1
        
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.cpp
#include <string>
#include <vector>
#include <iostream>


using namespace std;

int solution(vector<vector<string>> board, int h, int w) {

    int answer=0;
    
    cout << "board의 길이: " << board.size() << endl;
    
    vector<string> solv;
    

    
    
    if(h-1>=0) solv.push_back(board[h-1][w]);
    if(w-1>=0) solv.push_back(board[h][w-1]);
    if(w+1<board.size()) solv.push_back(board[h][w+1]);
    if(h+1<board.size()) solv.push_back(board[h+1][w]);
    
    if(!solv.empty()){
        for(int i=0; i<solv.size(); i++){
            if(solv[i]==board[h][w])
                answer++;
        }
    }
    
    
    
    return answer;

    
}